bin/update_cucumber

```diff
@@ -205,12 +230,15 @@ class {
             $nextVersion['hash'],
         );
 
-        // Remove `#` prefixes on issues and PR numbers (otherwise these will incorrectly autolink to our own repo).
-        $body = preg_replace('/#(\d+)/', '$1', $nextVersion['changelog']);
-        // Remove `@` prefixes on github username tags so that they don't create a notification / backlink
-        $body = preg_replace('/@(\w+)/', '$1', $body);
-        // Prevent any github links from spamming issues and PRs on other repos with backlinks
-        $body = preg_replace('_https://github\.com_', 'https://redirect.github.com', $body);
+        $replacements = [
+            // Remove `#` prefixes on issues and PR numbers (otherwise these will incorrectly autolink to our own repo).
+            '/#(\d+)/' => '$1',
+            // Remove `@` prefixes on github username tags so that they don't create a notification / backlink
+            '/@(\w+)/' => '$1',
+            // Prevent any github links from spamming issues and PRs on other repos with backlinks
+            '_https://github\.com_' => 'https://redirect.github.com',
+        ];
+        $body = preg_replace(array_keys($replacements), array_values($replacements), $nextVersion['changelog']);
```

Comment:
Is this refactor required for some phpstan reason? If not, personally I prefer it as individual statements.

Reply:
Kinda, since `preg_replace()` return `string|null` where $subject (of next call) requires `string`: https://phpstan.org/r/fca54ed7-62df-4da3-b3fb-7c0bb9ef7275

Another option is to typecast the result or the parameter:
```php
$body = (string) preg_replace('/#(\d+)/', '$1', $nextVersion['changelog']);
```

Let me know which one you prefer (although personally I'd prefer this approach - even better if PHP had a native `preg_translate` function).

Reply:
ah, OK. Let's stick with the array syntax then.